### PR TITLE
update tsconfig

### PR DESCRIPTION
**Pull request by srchase**
Updates tsconfig files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/tsconfig.es.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "lib": [
+      "es5",
+      "es2015.promise",
+      "es2015.collection",
+      "es2015.iterable",
+      "es2015.symbol.wellknown"
+    ],
+    "outDir": "dist/es"
+  }
+}
```
Comment (mtdowling) [inline smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/tsconfig.es.json]: Can you add a newline to this file?

File: smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/tsconfig.json [hunk 2]
```diff
@@ -25,6 +19,7 @@
     "hideGenerator": true,
     "ignoreCompilerErrors": true,
     "mode": "file",
-    "out": "./docs"
+    "out": "./docs",
+    "plugin": "@aws-sdk/client-documentation-generator"
   }
-}
+}
```
Comment (mtdowling) [inline smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/tsconfig.json]: Can you add a newline to this file?

Comment (srchase) [reply smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/tsconfig.json]: added.

File: smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/tsconfig.es.json (hunk 1 above)
Comment (srchase) [reply smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/tsconfig.es.json]: added.

**Pull request merged**